server/src-lib/Hasura/RQL/DDL/Schema/Cache.hs-boot

```diff
@@ -0,0 +1,20 @@
+module Hasura.RQL.DDL.Schema.Cache where
```

Comment:
There is no other way to get rid of cyclic dependencies?

Reply:
`SQL.DML` module was originally written as mutually recursive modules. I remember `hs-boot` files causing issues though I can't recollect exactly what the issues were. If it is easy to break the import cycle without resorting to `hs-boot` files we should give it try.

Reply:
I was wondering if this would come up. :)

There is an alternative strategy I had in mind that would avoid using `hs-boot` files by using typeclasses. Basically you use the typeclass as an interface and break the dependency by programming to that interface, then define an instance to effectively explicitly link the mutually recursive modules. I decided to go with `hs-boot` files instead, since to me it seems like a slightly easier solution (it’s basically doing the same thing but asking GHC to do the linking), but I understand why people like to avoid them. I’ll change it to use the typeclass approach if you’d prefer that.